bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/obs/plannedtime/PlannedTime.java

```diff
@@ -404,4 +420,67 @@ public static PlannedTime apply(Setup setup, List<Step> steps, ConfigSequence se
         steps = Collections.unmodifiableList(new ArrayList<>(steps));
         return new PlannedTime(setup, steps, sequence);
     }
+
+    /**
+     * The following methods are for use in the ITC.
+     *
+     */
+
+    public static final double VISIT_TIME = 7200; // visit time, sec
+    public static final double RECENTERING_INTERVAL = 3600; // sec
+
+    // total science time (time without setup and re-centering)
+    public long scienceTime() {
+        long scienceTime = 0;
+        for (Step step : steps) scienceTime += step.totalTime();
+        return scienceTime / 1000;
+    }
+
+    private int numReacq(double obsTime, double reacqInterval) {
+        int numReacq = 0;
+        if ((obsTime > reacqInterval) && (obsTime % reacqInterval == 0)) {
+            numReacq = (int) (obsTime / reacqInterval) - 1;
+        } else if ((obsTime > reacqInterval) && (obsTime % reacqInterval != 0)) {
+            numReacq = (int) (obsTime / reacqInterval);
+        }
+        return numReacq;
+    }
+
+    // number of acquisitions (setups) per observation
+    public int numAcq() {
+        return numReacq(scienceTime(), VISIT_TIME) + 1;
+    }
+
+    // Number of re-centerings on the slit for spectroscopic observations using PWFS2
+    public int numRecenter(Config config) {
+        int numRecenter = 0;
+        ItemKey guideWithPWFS2 = new ItemKey("telescope:guideWithPWFS2");
+
+        if (config.containsItem(guideWithPWFS2) &&
+                config.getItemValue(guideWithPWFS2).equals(StandardGuideOptions.Value.guide)) {
+            long scienceTime = scienceTime();
+            double visitTime = (numAcq() == 1) ? scienceTime : VISIT_TIME;
+            double lastVisitTime = scienceTime % VISIT_TIME;
+            // number of re-centerings per visit
+            int visitRecenteringNum = numReacq(visitTime, RECENTERING_INTERVAL);
+            // number of re-centerings in the last visit
+            int lastVisitRecenteringNum = numReacq(lastVisitTime, RECENTERING_INTERVAL);
+
+            if (VISIT_TIME > RECENTERING_INTERVAL) {
+                numRecenter = (numAcq() - 1) * visitRecenteringNum + lastVisitRecenteringNum;
+            } else {
+                throw new Error("Visit time is smaller than re-centering time");
+            }
+        }
+        return numRecenter;
+    }
+
+    // total time with acquisitions and re-acquisitions.
+    // Uses the parameter because of GSAOI LGS re-acquisitions.
+    public long totalTimeWithReacq(int numReacq) {
+        long totalTimeWithReacq = setup.time * numAcq() + setup.reacquisitionTime * numReacq;
+        for (Step step : steps) totalTimeWithReacq += step.totalTime();
+        return totalTimeWithReacq;
+    }
+
```

Comment:
I guess that these could probably be moved to a different class. I'll do that, at least.